presto-parquet/src/main/java/io/prestosql/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -229,6 +232,23 @@ public static Domain getDomain(Type type, long rowCount, Statistics<?> statistic
             return createDomain(type, hasNullValue, parquetIntegerStatistics);
         }
 
+        if (type.equals(TIMESTAMP) && statistics instanceof LongStatistics) {
+            LongStatistics longStatistics = (LongStatistics) statistics;
+            if (longStatistics.genericGetMin() > longStatistics.genericGetMax()) {
+                failWithCorruptionException(failOnCorruptedParquetStatistics, column, id, longStatistics);
+                return Domain.create(ValueSet.all(type), hasNullValue);
+            }
+
+            long min = longStatistics.getMin();
+            long max = longStatistics.getMax();
+
+            ParquetIntegerStatistics parquetIntegerStatistics =
+                    statistics.type().getOriginalType() == TIMESTAMP_MICROS
+                            ? new ParquetIntegerStatistics(MICROSECONDS.toMillis(min), MICROSECONDS.toMillis(max))
+                            : new ParquetIntegerStatistics(min, max);
```

Comment:
For `TIMESTAMP_MILLIS` I think do an explicit check here on original type because it looks like timestamp nanosecond support was added in https://github.com/apache/parquet-mr/pull/519 and it's using the new `LogicalTypeAnnotation` which will replace `OriginalType` in the next release of parquet-mr (see https://github.com/apache/parquet-mr/pull/463). So to be forward compatible if the check is explicit here it won't accidentally go into this path if a newer file is using nanoseconds.